### PR TITLE
Reddit Dataset Contribution

### PR DESCRIPTION
**Pull request by Jason-csc**
Add Reddit Dataset available from DGL with Node Classification Task and conveted it into GLI format.

### How Has This Been Tested?
Already run `make pytest DATASET=reddit` locally for Reddit dataset implementation test.

### CODE REVIEW

**Comment by Jason-csc** [comment]
`test_data_loading` and `test_node_classification_training` seem work on my computer.  Let me try to replicate the error locally and check it later.

**Comment by xingjian-zhang** [comment]
> `test_data_loading` and `test_node_classification_training` seem work on my computer. Let me try to replicate the error locally and check it later.

FYI, we use pytest for testing automation and it is convenient to run/debug tests locally in VSCode:

image

**Comment by Jason-csc** [comment]
Seems like when I use `gli.io.save_task_node_classification` to save `train_set`, `val_set` and `test_set` (int64 indices), it will first [convert them to list](https://github.com/Graph-Learning-Benchmarks/gli/blob/2302ac01fefeb421306d2d8891ad562102a6b5ac/gli/io/utils.py#L69) and then [to numpy array](https://github.com/Graph-Learning-Benchmarks/gli/blob/2302ac01fefeb421306d2d8891ad562102a6b5ac/gli/io/utils.py#L248).  The dtype numpy array will eventually become int32. 
Hence, when loading indices from npz file in pytest `test_data_loading` of dtype int32, `mask[_indices]` fails with _Error tensors used as indices must be long, byte or bool tensors_.
It works on my computer since my torch version == 2.1.1 while on server, torch version == 1.13.1

I'm trying to force the dtype of indices to be int64 when saving to node classification task npz file.

**Comment by xingjian-zhang** [comment]
Thanks @Jason-csc for the good catch. I am opening a new issue to fix this.

**Comment by xingjian-zhang** [comment]
Hi @Jason-csc, as #475 is merged, you may re-run your conversion script and see if everything is okay now. 🚀 Let us know if you find further issues.

**Comment by Jason-csc** [comment]
Oops, I think I've uploaded the task npz file to Dropbox. Maybe it need some time to synchronize ? 
Would you mind helping me have a look at that @xingjian-zhang ? Thx

**Comment by xingjian-zhang** [comment]
Yea it takes some time for the server to sync, I re-runed the test just now. Let's wait and see the results.

File: datasets/reddit/README.md [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+- Previous Version
+  + [Website](https://drive.google.com/drive/folders/1rq-H0XUM0BIRW9Pq5P4FMC9Xirpdx6zs)
+  + LICENSE: [MIT](https://github.com/GraphSAINT/GraphSAINT/blob/master/LICENSE)
+```
+@article{zeng2019graphsaint,
+  title={Graphsaint: Graph sampling based inductive learning method},
+  author={Zeng, Hanqing and Zhou, Hongkuan and Srivastava, Ajitesh and Kannan, Rajgopal and Prasanna, Viktor},
+  journal={arXiv preprint arXiv:1907.04931},
+  year={2019}
+}
```
Comment (xingjian-zhang) [inline datasets/reddit/README.md:48]: Not sure if it is okay to have previous version newer than current version. @jiaqima What do you think?

**Review by xingjian-zhang — COMMENTED** [review]
Others LGTM.

File: datasets/reddit/README.md (hunk 1 above)
Comment (Jason-csc) [reply datasets/reddit/README.md:48]: For the previous version, it's more like a variant (sparse version) for original Reddit Dataset. But the original version seems more widely used still. I have the same confusion with Xingjian and I'm open to get advice.

Comment (jiaqima) [reply datasets/reddit/README.md:48]: Where exactly does this version of dataset come from? If it comes from the 2017 version and has nothing to do with the 2019 paper, we should remove the 2019 paper. It's ok to leave previous versions blank.

Comment (Jason-csc) [reply datasets/reddit/README.md:48]: It comes from 2017 version. Let me remove 2019 paper for the previous version.

**Review by xingjian-zhang — APPROVED** [review]

**Review by jiaqima — APPROVED** [review]
LGTM

**Pull request merged**